sdk/go/common/util/gitutil/git.go

```diff
@@ -237,27 +251,152 @@ func getMatchedGroupsFromRegex(regex *regexp.Regexp, remoteURL string) map[strin
 	return groups
 }
 
-func parseAuthURL(remoteURL string) (string, *http.BasicAuth, error) {
-	u, err := url.Parse(remoteURL)
+// sshKeys memoizes keys we've loaded for given host URLs, to avoid needing to
+// re-fetch public keys.
+var sshKeys = map[string]transport.AuthMethod{}
```

Comment:
Should this be a `sync.Map` or have an accompanying mutex? 
Two concurrent calls to parseAuthURL will race to fill this otherwise.